### DISCUSSION TITLE
how can I add new attributes to the jsonPayload using micronaut-gcp-logging?

### DISCUSSION THREAD

**Discussion by glats** [micronaut-projects/micronaut-core#6568] [Q&A]
I want to add more attributes like "body" into the jsonPayload object like:
```
jsonPayload: {
    context: "default"
    logger: "me.jics.exception.handler.MappingExceptionHandler"
    message: "Error while mapping objects. Exception detail: Value not found"
    thread: "default-nioEventLoopGroup-1-2",
    body: {
        //... more attributes
    }
}
```

I see this field https://github.com/micronaut-projects/micronaut-gcp/blob/master/gcp-logging/src/main/java/io/micronaut/gcp/logging/StackdriverJsonLayout.java#L49 
but i don't know how to use it.
I tried:
```
<configuration>
    <appender name="CONSOLE_JSON" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
            <layout class="io.micronaut.gcp.logging.StackdriverJsonLayout">
                <projectId>${projectId}</projectId>
                <customJson>
                    {
                    "custom-key" : "custom-value"
                    }
                </customJson>
            </layout>
        </encoder>
    </appender>
    <root level="INFO">
        <appender-ref ref="CONSOLE_JSON" />
    </root>
</configuration>
```

But `customJson` is still null.

I saw that I can use `MDC.getMDCAdapter().put("test", "test")` and I can write in the jsonPayload.

My question: is it ok to use it like this?

**Comment by viniciusccarvalho** [comment]
Here's my 2c on this.

If you are trying to add cross cutting data to your logging system (maybe via an AOP, or a superclass) than that's more than fine IMO.

Unfortunately currently the customJson is not implemented, and it should be removed, my bad for leaving a trace of bad code there.

So what are the options:

1) User the MDC. As you pointed out you can use the MDC for this, and it should probably be fine, however bear in mind the MDC context is carried over threads, so anything on the path of your call will contain the keys you set there. Again, depending on your use case it could be the solution.

2) We could potentially add a way to pass custom values, but those will be static bound, so let's say you want to add the "custom-key" : "custom-value". A change would be:

```
<custom>
<entry>
<key>custom-key</key>
<value>custom-value</value>
<custom>
```
This would follow the logback joran's model of setting complex properties. But I honestly question this approach in lieu of using the first one, because the value would be printed for every call and its not dynamic
